test/extended/authentication.sh

```diff
@@ -65,28 +65,130 @@ wait_for_command 'oc get imagestream openldap --template="{{with \$tags := .stat
+	group2_ldapuid=$(awk 'NR == 2 {print $0}' ldapgroupuids.txt)
+	group3_ldapuid=$(awk 'NR == 3 {print $0}' ldapgroupuids.txt)
+
+	group1_osuid=$(awk 'NR == 1 {print $0}' osgroupuids.txt)
+	group2_osuid=$(awk 'NR == 2 {print $0}' osgroupuids.txt)
+	group3_osuid=$(awk 'NR == 2 {print $0}' osgroupuids.txt)
+
+	# create a temp copy of the test files 
+	cp * ${WORKINGDIR}
+	pushd ${WORKINGDIR} > /dev/null
+
+	# update sync-configs and validation files with the LDAP server's IP
+	config_files=sync-config*.yaml
+	validation_files=valid*.txt
+	for config in ${config_files} ${validation_files}
+	do
+		os::util::sed "s/LDAP_SERVICE_IP/${LDAP_SERVICE_IP}/g" ${config}
+	done
+
+	## TEST: Sync all LDAP groups from LDAP server ---------------------------------------------------------------##
+	openshift ex sync-groups \
+			--sync-config=sync-config.yaml --confirm || true
+	compare_and_cleanup valid_all_ldap_sync.txt
+
+	## TEST: Sync subset of LDAP groups from LDAP server using whitelist file ------------------------------------##
+	openshift ex sync-groups --whitelist=whitelist_ldap.txt \
+			--sync-config=sync-config.yaml --confirm
+	compare_and_cleanup valid_whitelist_ldap_sync.txt
+
+	## TEST: Sync subset of LDAP groups from LDAP server using literal whitelist ---------------------------------##
+	openshift ex sync-groups ${group1_ldapuid} --sync-config=sync-config.yaml --confirm
+	compare_and_cleanup valid_whitelist_ldap_sync.txt
+
+	## TEST: Sync subset of LDAP groups from LDAP server using union of literal whitelist and whitelist file -----##
+	openshift ex sync-groups ${group2_ldapuid} --whitelist=whitelist_ldap.txt \
+			--sync-config=sync-config.yaml --confirm
+	compare_and_cleanup valid_whitelist_union_ldap.txt
+
+	## TEST: Sync subset of OpenShift groups from LDAP server using whitelist file -------------------------------##
+	# sync group from LDAP
+	openshift ex sync-groups ${group1_ldapuid} \
+			--sync-config=sync-config.yaml --confirm
+	# add new member to the group in LDAP
+	ldapmodify -x -h ${LDAP_SERVICE_IP} -p 389 -D cn=Manager,dc=example,dc=com -w admin -f modify_group1.ldif
+	# sync some groups from whitelist file from OpenShift
+	openshift ex sync-groups --type=openshift --whitelist=whitelist_openshift.txt \
+			--sync-config=sync-config.yaml --confirm
+	compare_and_cleanup valid_whitelist_openshift_sync.txt
+	# undo changes to LDAP server
+	ldapmodify -x -h ${LDAP_SERVICE_IP} -p 389 -D cn=Manager,dc=example,dc=com -w admin -f unmodify_group1.ldif
+
+	## TEST: Sync subset of OpenShift groups from LDAP server using literal whitelist ----------------------------##
+	# sync group from LDAP
+	openshift ex sync-groups ${group1_ldapuid} \
+			--sync-config=sync-config.yaml --confirm
+	# add new member to the group in LDAP
+	ldapmodify -x -h ${LDAP_SERVICE_IP} -p 389 -D cn=Manager,dc=example,dc=com -w admin -f modify_group1.ldif
```

Comment:
In general, ldap changes will be harder for future maintainers to read.  Manipulation openshift resources will be easier.  We have the client tools for cases of dire need, but otherwise, prefer to manipulate the openshift groups using `oadm groups --help` commands.